### DISCUSSION TITLE
pip installing BASTA as a developer

### DISCUSSION THREAD

**Discussion by amaliestokholm** [BASTAcode/BASTA#42] [Show and tell]
We are going to make the switch to an easier installation using `pip`.
However, if you are a developer - how do you install `BASTA` using `pip` but still keep it working with `git`?

It is rather simple - here is one way:
- Firstly, remember to remove/comment out the old `BASTA`-lines in your .bashrc file from your old installation.
- Then go to your `BASTA` directory. In principle, this is not scriptly nessesairy as you can set-up the virtual environment anywhere, but for simplicity and structure this is a good choice.
- Remember to `git fetch`, `git pull`, and (at the time of this publication) to `git checkout` the `devel` branch.
- Now, rename or remove your old `BASTA` virtuel environment
  ```
  mv venv/ old_venv/
  ```
- Now create a new virtual environment
	```
	python3 -m venv venv	
	```
- Source it
	```
	source venv/bin/activate
	```
- Now the magic happens. We pip install **the git repository** with `BASTA`.
	```
	pip install -e .
	```
If you are creating the virtual environment not in your `BASTA` directory, you instead need to
```
	pip install -e path/to/git/checkout
```

And that should be it for now.
If this is not an update from an old installation to a new one, you might need to run `BASTAdownload` to help set-up the last couple of things, but otherwise you should be good to go ;)

**Comment by jakobmoss** [comment]
This is the approach currently in the documentation (on devel, https://basta.readthedocs.io/en/devel/install.html) although in a more compact form. That page assumes a clean install, of course.
